backend/api/events.py

```diff
@@ -0,0 +1,145 @@
+    try:
+        # Return created event
+        return event_service.create(subject, event)
+    except Exception as e:
+        # Raise 422 exception if creation fails
+        # - This would occur if the request body is shaped incorrectly
+        raise HTTPException(status_code=422, detail=str(e))
+    
+@api.get("/{id}", responses={404: {"model": None}}, response_model=EventDetails, tags=['Events'])
+def get_event_from_id(id: int, event_service: EventService = Depends()) -> EventDetails:
+    """
+    Get event with matching id
+
+    Parameters:
+        id: an int representing a unique Event ID
+        event_service: a valid EventService
+    
+    Returns:
+        EventDetails: a valid EventDetails model corresponding to the given event id
+    
+    Raises:
+        HTTPException 404 if update() raises an Exception
+    """
+    
+    # Try to get event with matching id
+    try: 
+        # Return event
+        return event_service.get_from_id(id)
+    except (EventNotFoundException, UserPermissionException) as e:
+        # Raise 404 exception if search fails
+        # - This would occur if there is no response
+        raise HTTPException(status_code=404, detail=str(e))
+
+@api.put("", responses={404: {"model": None}}, response_model=EventDetails, tags=['Events'])
+def update_event(event: EventDetails, subject: User = Depends(registered_user), event_service: EventService = Depends()) -> EventDetails:
+    """
+    Update event
+
+    Parameters:
+        event: a valid Event model
+        subject: a valid User model representing the currently logged in User
+        event_service: a valid EventService
+
+    Returns:
+        EventDetails: a valid EventDetails model representing the updated Event
+
+    Raises:
+        HTTPException 404 if update() raises an Exception
+    """
+
+    try: 
+        # Return updated event
+        return event_service.update(subject, event)
+    except (EventNotFoundException, UserPermissionException) as e:
+        # Raise 404 exception if search fails
+        # - This would occur if there is no response
+        raise HTTPException(status_code=404, detail=str(e))
+
+@api.delete("/{id}", tags=['Events'])
+def delete_event(id: int, subject: User = Depends(registered_user), event_service = Depends(EventService)):
+    """
+    Delete event based on id
+
+    Parameters:
+        id: an int representing a unique event ID
+        subject: a valid User model representing the currently logged in User
+        event_service: a valid EventService
+    
+    Raises:
+        HTTPException 404 if delete() raises an Exception
+    """
+   
+    try:
+        # Try to delete event
+        event_service.delete(subject, id)
+    except (EventNotFoundException, UserPermissionException) as e:
+        # Raise 404 exception if search fails
+        # - This would occur if there is no response or if item to delete does not exist
+        raise HTTPException(status_code=404, detail=str(e))
```

Comment:
Comments again redundant. Here, I think the right move for `UserPermission` is just to let the application-level middleware handle it.